app/src/main/java/com/paddlesandbugs/dahdidahdit/brasspound/AbstractMorseInput.java

```diff
@@ -121,8 +124,10 @@ public void init(Decoder.CharListener charListener) {
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
         final int keyCodeLeft = prefs.getInt(PREFS_KEY_KEY_CODE_LEFT, KeyEvent.KEYCODE_A);
         final int keyCodeRight = prefs.getInt(PREFS_KEY_KEY_CODE_RIGHT, KeyEvent.KEYCODE_B);
+        final boolean isMouseLeft = prefs.getBoolean(PREFS_KEY_IS_MOUSE_LEFT, false);
+        final boolean isMouseRight = prefs.getBoolean(PREFS_KEY_IS_MOUSE_RIGHT, false);
```

Comment:
Looking at how OnScreenPaddle gets a context passed, it's probably time to pass a context to HardwarePaddle and move the preference-pulling there.

Reply:
Indeed! I didn't want to overload this PR, but I think some code is quite loosely coupled and would benefit from a utility class or method on the prefs.